### DISCUSSION TITLE
Doubts regarding webhooks

### DISCUSSION THREAD

**Discussion by slice-saurabha** [gotenberg/gotenberg#590] [Q&A] [answered]
Hi guys,

I've a few doubts about the use of webhooks.

1) Does the webhook create the file locally before uploading it somewhere else or it keeps it in-memory?
2) If it does create the file locally, does it delete it if the upload happens or if the upload fails?

**Comment by gulien** [comment, marked as answer]
Hello @slice-saurabha,

The webhook feature allows your to upload the resulting PDF to a given destination instead of returning it to the client as a response.
Gotenberg holds the file(s) only for the duration of the request or a timeout.

**Reply by slice-saurabha** [reply]
Hi @gulien , why don't you guys support returning pdf in the response. Lets say if I've a requirement of doing something with the generated pdf, then I've to basically write it on the disk first and then read it back. It adds extra cost of disk io.

Just trying to understand the thought process behind this decision.

**Reply by gulien** [reply]
What do you mean by "returning the pdf in the response"?

**Reply by slice-saurabha** [reply]
My bad. I misunderstood. Thanks for the quick replies bro :)
